homeassistant/components/lcn/climate.py

```diff
@@ -0,0 +1,139 @@
+"""Support for LCN climate control."""
+
+from homeassistant.components.climate import ClimateDevice, const
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_ADDRESS, CONF_UNIT_OF_MEASUREMENT)
+
+from . import LcnDevice, get_connection
+from .const import (
+    CONF_CONNECTIONS, CONF_LOCKABLE, CONF_MAX_TEMP, CONF_MIN_TEMP,
+    CONF_SETPOINT, CONF_SOURCE, DATA_LCN)
+
+DEPENDENCIES = ['lcn']
+
+
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Set up the LCN climate platform."""
+    if discovery_info is None:
+        return
+
+    import pypck
+
+    devices = []
+    for config in discovery_info:
+        address, connection_id = config[CONF_ADDRESS]
+        addr = pypck.lcn_addr.LcnAddr(*address)
+        connections = hass.data[DATA_LCN][CONF_CONNECTIONS]
+        connection = get_connection(connections, connection_id)
+        address_connection = connection.get_address_conn(addr)
+
+        devices.append(LcnClimate(config, address_connection))
+
+    async_add_entities(devices)
+
+
+class LcnClimate(LcnDevice, ClimateDevice):
+    """Representation of a LCN climate device."""
+
+    def __init__(self, config, address_connection):
+        """Initialize of a LCN climate device."""
+        super().__init__(config, address_connection)
+
+        self.variable = self.pypck.lcn_defs.Var[config[CONF_SOURCE]]
+        self.setpoint = self.pypck.lcn_defs.Var[config[CONF_SETPOINT]]
+        self.unit = self.pypck.lcn_defs.VarUnit.parse(
+            config[CONF_UNIT_OF_MEASUREMENT])
+
+        self.regulator_id = \
+            self.pypck.lcn_defs.Var.to_set_point_id(self.setpoint)
+        self.is_lockable = config[CONF_LOCKABLE]
+        self._max_temp = config[CONF_MAX_TEMP]
+        self._min_temp = config[CONF_MIN_TEMP]
+
+        self._current_temperature = None
+        self._target_temperature = None
+        self._is_on = True
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added to hass."""
+        await super().async_added_to_hass()
+        self.hass.async_create_task(
```

Comment:
Do we need to create a task or can we await this?

Reply:
We can await this. No need for another task object.
For the other plaforms see also #23321